helper/bearer.go

```diff
@@ -25,9 +25,28 @@ import (
 	"strings"
 )
 
-func BearerTokenFromRequest(r *http.Request) string {
-	auth := r.Header.Get("Authorization")
-	split := strings.SplitN(auth, " ", 2)
+const (
+	defaultAuthorizationHeader = "Authorization"
+)
+
+type BearerTokenLocation struct {
+	Header         *string `json:"header"`
+	QueryParameter *string `json:"query_parameter"`
+}
+
+func BearerTokenFromRequest(r *http.Request, tokenLocation *BearerTokenLocation) string {
```

Comment:
Would it make sense to test this independently?

Reply:
Yeah. That might be a good idea :)
I added tests.